### PR TITLE
Add search form to Admin models page

### PR DESCRIPTION
**Pull request by lehors**

### CODE REVIEW

File: web/modules/mof/src/Form/ModelSearchForm.php [hunk 1]
```diff
@@ -30,9 +30,11 @@ public function getFormId() {
   /**
    * {@inheritdoc}
    */
-  public function buildForm(array $form, FormStateInterface $form_state) {
+  public function buildForm(array $form, FormStateInterface $form_state, string $collection_name = NULL) {
     $req = $this->request;
 
+    $form['collection'] = $collection_name;
```
Comment (lehors) [inline web/modules/mof/src/Form/ModelSearchForm.php:36]: @gregcube This works but may not be the best way to pass this around. Any suggestion to do that better welcome! Thanks.

Comment (gregcube) [reply web/modules/mof/src/Form/ModelSearchForm.php:36]: I'll take a look and follow up soon.

Comment (gregcube) [reply web/modules/mof/src/Form/ModelSearchForm.php]: We don't need to pass $collection_name at all.
We can pull the route name from the symfony request object.

**Comment by gregcube** [comment]
I've made two small changes:

1) Removed $collection_name; we'll get the route name from the request.

2) Added cache contexts to the list builders.  I've enabled caching with the API addition to speed up response times -- both for API traffic and regular web traffic.  These cache contexts ensure that cached data is properly invalidated and updated based on specific contextual changes.

**Comment by lehors** [comment]
> I've made two small changes:
> 
>     1. Removed $collection_name; we'll get the route name from the request.

Great. I'm learning more and more about Drupal but I must admit that this kind of stuff is really hard to figure out. Even with the answer I can't easily find a piece of documentation that would tell me how to get that info. :-/
 
>     2. Added cache contexts to the list builders.  I've enabled caching with the API addition to speed up response times -- both for API traffic and regular web traffic.  These cache contexts ensure that cached data is properly invalidated and updated based on specific contextual changes.

Sounds good.
Thank you!

**Pull request merged**